File: Problems/557_Reverse_Words_in_a_String_III.cpp
#include<string>
using namespace std;

class Solution {
public:
    string reverseWords(string s) {
        int start = 0;
        int end = 0;
        string res;
        for(int i=0; i<s.length(); i++){
            if(s[i]==' '){
                end = i;
                for(int j=i-1; j>=start; j--){
                    res += s[j];
                }
                res+=' ';
                start = i+1;
            }
        }
        for(int j=s.length()-1; j>= start; j--){
            res += s[j];
        }
        return res;
    }
};

File: Problems/557_Reverse_Words_in_a_String_III.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s_list = s.split(' ')
        res = ""
        for word in s_list:
            for i in range(len(word)-1, -1, -1):
                res += word[i]
            res +=' '
        return res[:-1]